pkg/wcore/workspace.go

```diff
@@ -114,40 +115,58 @@ func UpdateWorkspace(ctx context.Context, workspaceId string, name string, icon
 // If force is true, it will delete even if workspace is named.
 // If workspace is empty, it will be deleted, even if it is named.
 // Returns true if workspace was deleted, false if it was not deleted.
-func DeleteWorkspace(ctx context.Context, workspaceId string, force bool) (bool, error) {
+func DeleteWorkspace(ctx context.Context, workspaceId string, force bool) (bool, string, error) {
 	log.Printf("DeleteWorkspace %s\n", workspaceId)
 	workspace, err := wstore.DBMustGet[*waveobj.Workspace](ctx, workspaceId)
 	if err != nil {
-		return false, fmt.Errorf("error getting workspace: %w", err)
+		return false, "", fmt.Errorf("error getting workspace: %w", err)
 	}
 	if workspace.Name != "" && workspace.Icon != "" && !force && (len(workspace.TabIds) > 0 || len(workspace.PinnedTabIds) > 0) {
 		log.Printf("Ignoring DeleteWorkspace for workspace %s as it is named\n", workspaceId)
-		return false, nil
+		return false, "", nil
 	}
 
 	// delete all pinned and unpinned tabs
 	for _, tabId := range append(workspace.TabIds, workspace.PinnedTabIds...) {
 		log.Printf("deleting tab %s\n", tabId)
 		_, err := DeleteTab(ctx, workspaceId, tabId, false)
 		if err != nil {
-			return false, fmt.Errorf("error closing tab: %w", err)
+			return false, "", fmt.Errorf("error closing tab: %w", err)
 		}
 	}
 	windowId, err := wstore.DBFindWindowForWorkspaceId(ctx, workspaceId)
 	err = wstore.DBDelete(ctx, waveobj.OType_Workspace, workspaceId)
 	if err != nil {
-		return false, fmt.Errorf("error deleting workspace: %w", err)
+		return false, "", fmt.Errorf("error deleting workspace: %w", err)
 	}
 	log.Printf("deleted workspace %s\n", workspaceId)
 	wps.Broker.Publish(wps.WaveEvent{
-		Event: wps.Event_WorkspaceUpdate})
+		Event: wps.Event_WorkspaceUpdate,
+	})
+
 	if windowId != "" {
-		err = CloseWindow(ctx, windowId, false)
+		workspaces, err := ListWorkspaces(ctx)
+		if err != nil {
+			return false, "", fmt.Errorf("error retrieving workspaceList: %w", err)
+		}
+		UnclaimedWorkspace := ""
+		for _, ws := range workspaces {
```

Comment:
I liked your old logic which would attempt to find the workspace at index closest to the one that was deleted, could you incorporate that here?

I think you can do this in one pass. Create three variables, `curWsIndex`, `freeWsIndexBeforeCurrent` and `freeWsIndexAfterCurrent`. Then, you iterate through the workspace list, setting `freeWsIndexBeforeCurrent` to the index in the workspace list where you find a free workspace (not claimed by a window). Once you find the current index, you set `curWsIndex` and from there you continue iterating until you find another free workspace, where you set `freeWsIndexAfterCurrent`. Then, you just take whichever one of the two indices is closer to `curWsIndex` (or whichever isn't null).

Reply:
sure i can implement that if we can ignore this return error, because im relying on getting any claimable workspace to not upset this code because deleteworkspace also runs in switchworkspace for some reason i dont know
![image](https://github.com/user-attachments/assets/e4c10836-ea9a-43e7-b0c9-4bba7f60eb08)

Reply:
Ah it runs in switch workspace to clean up unsaved workspaces. For saved ones, delete without a force flag is a no-op

Reply:
how do I actually retrieve the error string in the error data structure?  if I can get the string to reliably know if 'not found' happened I wont have to limit the claimworkspace implementation, I only saw msg inside of it but couldn't access it, sorry I'm new to go.

Reply:
I think you can return empty string if nothing is deleted. This should never happen if you call deleteworkspace from the workspace service since that passes the force flag as true to the wcore implementation

Reply:
OK ill implement the old logic tomorrow and notify you, ty!